components/ILIAS/Certificate/classes/Form/Repository/class.ilCertificateSettingsFormRepository.php

```diff
@@ -185,33 +195,37 @@ public function createForm(ilCertificateGUI $certificateGUI): ilPropertyFormGUI
-        if (!$this->backGroundImageFileService->hasBackgroundImage($certificateTemplate)) {
-            if ($this->global_certificate_settings->hasBackgroundImage()) {
-                ilWACSignedPath::setTokenMaxLifetimeInSeconds(15);
-                $imagePath = ilWACSignedPath::signFile($this->global_certificate_settings->getBackgroundImageThumbPathWeb());
-                $bgimage->setImage($imagePath);
-                $bgimage->setAllowDeletion(false);
-            }
-        } else {
-            ilWACSignedPath::setTokenMaxLifetimeInSeconds(15);
-
-            $thumbnail_path = $this->backGroundImageFileService->getBackgroundImageThumbPath();
-
-            if (!is_file($thumbnail_path)) {
-                //Trying if it uses default image path
-                $thumbnail_path = CLIENT_WEB_DIR . $certificateTemplate->getBackgroundImagePath() . '.thumb.jpg';
-                if (!is_file($thumbnail_path)) {
-                    //Trying to use global default image
-                    $thumbnail_path = $this->global_certificate_settings->getDefaultBackgroundImageThumbPath();
-                    if (!is_file($thumbnail_path)) {
-                        //No image global default configured
-                        $thumbnail_path = '';
-                    }
-                }
-                $bgimage->setALlowDeletion(false);
+        $bg_image_rid = $certificateTemplate->getBackgroundImageIdentification();
+        $bg_image_path = $certificateTemplate->getBackgroundImagePath();
+        if (
+            (
+                $this->global_certificate_settings->getBackgroundImageIdentification() instanceof ResourceIdentification &&
+                $bg_image_rid === $this->global_certificate_settings->getBackgroundImageIdentification()->serialize()
+            ) ||
+            !$certificateTemplate->getBackgroundImageIdentification() ||
+            !$this->irss->manage()->find(
+                $certificateTemplate->getBackgroundImageIdentification()
+            ) instanceof ResourceIdentification
+        ) {
+            $bgimage->setAllowDeletion(false);
+        }
+        $identification = $this->irss->manage()->find($certificateTemplate->getBackgroundImageIdentification());
+        if ($identification instanceof ResourceIdentification) {
+            $background_flavour = $this->irss->flavours()->get(
+                $identification,
+                $this->card_thumbnail_definition
+            );
+            $flavour_urls = $this->irss->consume()->flavourUrls($background_flavour);
+            foreach ($flavour_urls->getURLs() as $url) {
+                /** @var string $url */
+                $bgimage->setImage($url);
             }
-            $imagePath = ilWACSignedPath::signFile($thumbnail_path);
-            $bgimage->setImage($imagePath);
+        } elseif ($this->filesystem->has($bg_image_path)) {
+            $bgimage->setImage(
+                ilWACSignedPath::signFile(
```

Comment:
`public` will IMO not be part of the HTTP path, if ILIAS is configured correctly.